### DISCUSSION TITLE
How to disable rust-analyzer cargo check alltargets

### DISCUSSION THREAD

**Discussion by XiaoPengYouCode** [zed-industries/zed#21270] [Feature Requests]
I an using zed to develop a lib use embassy embedded framework, I wanna konw how to disable the rust-analyzer cargo check alltargets options

in vscode I can use this option

```json
// ./.vscode/settings.json
{
    "rust-analyzer.cargo.allTargets": false
}
```

I try it the same way in zed

```json
// ./.zed/settings.json
{
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "cargo": {
          "allTargets": false
        }
      }
    }
  }
}
```

How to solve this problem?

**Comment by piedoom** [comment]
hey xiaopengyou (小朋友?) - ran into the same issue. I found this fix which appears to work on my end. I had to restart zed for changes to take effect.

```
{
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "check": {
          "allTargets": false
        }
      }
    }
  }
}
```

hope that helps! (full credit to the [stm32f4xx-hal](https://github.com/stm32-rs/stm32f4xx-hal/blob/ed88ea13073d6497d95c8f888e7264cbb648ef4b/.zed/settings.json#L9) crate which I lifted this from)

**Reply by XiaoPengYouCode** [reply]
I have decided to switch to vscode to continue development, mainly because the probe-rs extension is more complete in vscode and does not have these strange problems, but I don’t know why when I opened zed again today, whether it is the check or cargo option, disable alltargets It all works for me, so weird

我已经决定切换vscode继续开发了，主要是由于probe-rs扩展在vscode较为完善，并且没有这些怪问题，但是我不知道为什么当我今天再一次打开zed，无论是check还是cargo选项内，disable alltargets对我都适用了，好怪

Thanks for your reply💖

感谢回复💖

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
